### PR TITLE
YD-652 Add app version info to logging context

### PR DESCRIPTION
**Pull request by Bert-R**
Added the information from the Yona-App-Version header to the MDC.

Along with this corrected the message IDs of the Dutch messages for PIN reset confirmation codes from "error.pin.reset.overwrite.confirmation" into "error.pin.reset.request.confirmation".

### CODE REVIEW

File: core/src/main/java/nu/yona/server/rest/ErrorLoggingFilter.java [hunk 1]
```diff
@@ -72,6 +86,52 @@ private static String getCorrelationId(HttpServletRequest request)
...
+		{
+			Optional.ofNullable(request.getHeader(Constants.APP_VERSION_HEADER))
+					.ifPresent(LoggingContext::validateAppVersionHeader);
+		}
+
+		private static void validateAppVersionHeader(String header)
+		{
+			String[] parts = header.split("/");
+			Require.that(parts.length == 3, () -> InvalidDataException.invalidAppVersionHeader(header));
+			assertValidOperatingSystem(parts[0]);
```
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/rest/ErrorLoggingFilter.java]: This puts quite strict requirements on the app version header. Can we test this in time that it does not block the app?

File: core/src/main/java/nu/yona/server/rest/ErrorLoggingFilter.java [hunk 2]
```diff
@@ -72,6 +86,52 @@ private static String getCorrelationId(HttpServletRequest request)
 			return traceIdHeader;
 		}
 
+		private static void putAppVersionContext(String header)
+		{
+			String[] parts = header.split("/");
+			String operatingSystem = parts[0];
+			int versionCode = Integer.parseInt(parts[1]);
+			String versionName = parts[2];
+			MDC.put(APP_OS_MDC_KEY, operatingSystem);
```
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/rest/ErrorLoggingFilter.java]: Have we tested this?

File: core/src/main/java/nu/yona/server/rest/ErrorLoggingFilter.java (hunk 1 above)
Comment (Bert-R) [reply core/src/main/java/nu/yona/server/rest/ErrorLoggingFilter.java]: Yes, on the beta server

File: core/src/main/java/nu/yona/server/rest/ErrorLoggingFilter.java (hunk 2 above)
Comment (Bert-R) [reply core/src/main/java/nu/yona/server/rest/ErrorLoggingFilter.java]: Yes, through the Groovy test

**Pull request merged**